crates/pera-core/src/consensus_handler.rs

```diff
@@ -355,6 +355,63 @@ impl<C: CheckpointServiceNotify + Send + Sync> ConsensusHandler<C> {
                             .stats
                             .inc_num_user_transactions(authority_index as usize);
                     }
+
+                    // If we receive a `DwalletMPCOutput` transaction,
+                    // verify that it's valid and create a system transaction
+                    // to store its output on the blockchain,
+                    // so it will be available for the initiating user.
+                    if let ConsensusTransactionKind::DwalletMPCOutput(
+                        value,
+                        session_id,
+                        initiating_address,
+                    ) = &transaction.kind
+                    {
+                        info!(
+                            "Received proof mpc output from authority {:?} for session {:?}",
+                            authority_index, session_id
+                        );
+
+                        let mpc_manager = self.epoch_store.proof_mpc_manager.get();
+                        let is_valid_transaction = match mpc_manager {
+                            Some(mpc_manager) => {
+                                mpc_manager.lock().await
+                                    .try_verify_output(value, session_id)
```

Comment:
Correct me if I'm wrong, but this code only checks that the output contained in the `DWalletMPCOutput` transaction is equal to the local output. So this function doesn't measure whether this specific output value has received a quorum of votes from other validators.

Reply:
Ok; but a "system tx" is created inside the local Move VM of the validator; not inside the consensus, right? So a validator cannot see how many other validators created the same system tx inside their Move VM. Or am I misunderstanding something?

Reply:
Ye you understand right, but this system TX is being committed to the chain only if a quorum of validators agree on it, otherwise a fork in the chain will be created